src/io/flutter/devtools/DevToolsManager.java

```diff
@@ -311,6 +353,31 @@ public void openBrowserAndConnect(String serviceProtocolUri, String page) {
       null
     );
   }
+
+  public void openPanel(String serviceProtocolUri, ContentManager contentManager, String tabName) {
+    String address = DevToolsUtils.generateDevToolsUrl(devtoolsHost, devtoolsPort, serviceProtocolUri, null) + "&embed=true&page=inspector";
+
+    EngineOptions options =
+      EngineOptions.newBuilder(HARDWARE_ACCELERATED).build();
+    Engine engine = Engine.newInstance(options);
+    Browser browser = engine.newBrowser();
+
+    final Content content = contentManager.getFactory().createContent(null, tabName, false);
+
+    SwingUtilities.invokeLater(() -> {
```

Comment:
why the invoke later? I would have thought this code was already running on the ui thread.

Reply:
I'm not sure how to tell, actually. This is called from a process listening for devtools to start, is that on a separate thread?

Reply:
an easy way to tell is to set a breakpoint and see what thread you are on or add an assert that you are on the UI thread.